### PR TITLE
Add sky and sun matrix

### PR DESCRIPTION
**Pull request by mostaphaRoudsari**
This PR adds skymtx and sunmtx classes. I decided not to create a daylightmtx class as the separation is actually really helpful to keep the `to_radiance` method clean and clear.

The PR also improves documentation of other sky modules.

### CODE REVIEW

**Comment by chriswmackey** [comment]
Thanks @mostaphaRoudsari . I will review this shortly. Looking forward to implementing the grasshopper components for these once they are merged.

File: honeybee_radiance/lightsource/sky/sunmatrix.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+        A number between -360 and 360 for the counterclockwise difference between
+        the North and the positive Y-axis in degrees. 90 is West and 270 is East.
+        """
+        return self._north
+
+    @north.setter
+    def north(self, value):
+        value = typing.float_in_range(value, -360, +360, 'Skymatrix north')
+        self._north = value
```
Comment (chriswmackey) [inline honeybee_radiance/lightsource/sky/sunmatrix.py:62]: This is just a recommendation but, if you want to expose a method to set the north using a vector, it's very easy to do with ladybug_geometry.  You can see some sample code here on the honeybee-core Model:
https://github.com/ladybug-tools/honeybee-core/blob/master/honeybee/model.py#L238-L239

File: honeybee_radiance/lightsource/sky/sunmatrix.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+        Note that you need to write the wea to a file (in.wea) before running this
+        command.
+
+        Alternatively you can use write method which will write the wea data to a file.
+
+        Args:
+            output_type: An integer between 0 to 2 for output type.
+                * 0 = output in W/m2/sr visible
+                * 1 = output in W/m2/sr solar (default)
```
Comment (chriswmackey) [inline honeybee_radiance/lightsource/sky/sunmatrix.py:116]: If this is `an integer between 0 and 2`, then what does `2` signify? 

If 2 doesn't mean anything and this is only ever two options (visible and solar), maybe a boolean input is more appropriate.

File: honeybee_radiance/lightsource/sky/sunmatrix.py [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+    def to_file(self, folder, name=None, hoys=None, mkdir=False):
+        """Write matrix to a Radiance file.
+
+        This method also writes the wea information to a wea file.
+
+        Args:
+            folder: Target folder.
+            name: File name.
+            mkdir: A boolean to note if the directory should be created if doesn't
```
Comment (chriswmackey) [inline honeybee_radiance/lightsource/sky/sunmatrix.py:167]: You are missing a description of `hoys` in the docstring here.

File: honeybee_radiance/lightsource/sky/sunmatrix.py [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+        """
+        name = typing.valid_string(name) if name \
+            else '%s.rad' % self.__class__.__name__.lower()
+        # write wea file first
+        wea_file = self.write_wea(folder, hoys=hoys)
+        content = self.to_radiance(wea_file=os.path.split(wea_file)[-1])
+        return futil.write_to_file_by_name(folder, name, '!' + content, mkdir)
+
+    def __eq__(self, value):
+        if type(value) != type(self) \
```
Comment (chriswmackey) [inline honeybee_radiance/lightsource/sky/sunmatrix.py:181]: Isn't it a little cleaner to use:
```
if isinstance(other, self.__class__)
```

File: honeybee_radiance/lightsource/sky/skymatrix.py [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+            output_name: A name for output files (default: sky_mtx).
+            cumulative: A boolean to generate cumulative sky. This option is only
+                available in Radiance 5.3 and higher versions (default: False).
+            components: An integer between 0-2 to note the distribution of which
+                components should be included. 0 might be used to include both solar and
+                sky contribution. 1 may be used to produce a sun-only matrix, with no sky
+                contributions.  Alternatively, 2 may be used to exclude any direct solar
+                component from the output.  If there is a sun in the description,
+                gendaymtx will include its contribution in the four nearest sky patches,
+                distributing energy according to centroid proximity (default: 0).
```
Comment (chriswmackey) [inline honeybee_radiance/lightsource/sky/skymatrix.py:122]: I feel like some of the vocabulary here can be clearer or more consistent. Is your use of the word `solar` and `sun` interchangeable? Maybe it's just better to stick to using `sun`. It's also a little confusing that sometimes, you include the word `direct solar` but other times, you just say `solar`. Are these two the same?

File: honeybee_radiance/lightsource/sky/sunmatrix.py (hunk 1 above)
Comment (mostaphaRoudsari) [reply honeybee_radiance/lightsource/sky/sunmatrix.py:62]: For now I keep it as a number and leave the vector based input conversion to numbers to each interface.

File: honeybee_radiance/lightsource/sky/sunmatrix.py (hunk 2 above)
Comment (mostaphaRoudsari) [reply honeybee_radiance/lightsource/sky/sunmatrix.py:116]: Boolean is an option but using an integer for output_type will keep it consistent among different skies.

**Pull request merged**